// tokend-sdk/lib/const/enums.const.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OP_TYPES = exports.ACCOUNT_TYPES = exports.ASSET_PAIR_POLICIES = exports.ASSET_POLICIES = exports.PAYMENT_FEE_SUBTYPES = exports.FEE_TYPES = exports.REQUEST_TYPES = undefined;
exports.xdrEnumToConstant = xdrEnumToConstant;

var _base = require('../base');

function xdrEnumToConstant(xdrEnum) {
  xdrEnum = typeof xdrEnum === 'string' ? _base.xdr[xdrEnum] : xdrEnum;

  try {
    const res = {};
    xdrEnum.values().forEach(function (item) {
      res[item.name] = item.value;
    });
    return res;
  } catch (error) {
    throw new Error(`xdrEnumToConstant: Cannot get values from provided xdrEnum (${xdrEnum})`);
  }
}

const REQUEST_TYPES = exports.REQUEST_TYPES = Object.freeze(xdrEnumToConstant('ReviewableRequestType'));

const FEE_TYPES = exports.FEE_TYPES = Object.freeze(xdrEnumToConstant('FeeType'));

const PAYMENT_FEE_SUBTYPES = exports.PAYMENT_FEE_SUBTYPES = Object.freeze(xdrEnumToConstant('PaymentFeeType'));

const ASSET_POLICIES = exports.ASSET_POLICIES = Object.freeze(xdrEnumToConstant('AssetPolicy'));

const ASSET_PAIR_POLICIES = exports.ASSET_PAIR_POLICIES = Object.freeze(xdrEnumToConstant('AssetPairPolicy'));

const ACCOUNT_TYPES = exports.ACCOUNT_TYPES = Object.freeze(xdrEnumToConstant('AccountType'));

const OP_TYPES = exports.OP_TYPES = Object.freeze(xdrEnumToConstant('OperationType'));
